src/compress.rs

```diff
@@ -27,22 +27,22 @@ pub enum Compressor {
 }
 
 impl Compressor {
-    const LZ4_ID: &'static [u8; 8] = b"lz4     ";
-    const ZSTD_ID: &'static [u8; 8] = b"zstd    ";
+    const LZ4_MAGIC_NUMBER: u32 = 0x184D2204;
+    const ZSTD_MAGIC_NUMBER: u32 = 0xFD2FB528;
```

Comment:
I would really recommend encoding these as byte-strings rather than `u32` because endianness.

Reply:
```suggestion
    const LZ4_MAGIC_NUMBER: u32 = b"\x18\x4D\x22\x04";
    const ZSTD_MAGIC_NUMBER: u32 = b"\xFD\x2F\xB5\x28";
```
and then you don't have to deal with endian conversions.

Reply:
Done